src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+					ShowSevereError( "Autosizing of water flow requires a heating loop Sizing:Plant object" );
+					ShowContinueError( "Occurs in " + this->unitType + " Object=" + this->name );
+					ErrorsFound = true;
+				} else {
+					this->hWTempIn = DataSizing::PlantSizData(pltSizHeatNum).ExitTemp;
+				}
+				this->mDotCW = 0.0;
+				this->tDBZoneAirTemp = FinalZoneSizing( CurZoneEqNum ).ZoneTempAtHeatPeak;
+				this->tDBSystemAir = FinalZoneSizing( CurZoneEqNum ).DesHeatCoilInTempTU;
+				this->cpZoneAir = PsyCpAirFnWTdb(	DataSizing::FinalZoneSizing( CurZoneEqNum ).ZoneHumRatAtHeatPeak, 
+													DataSizing::FinalZoneSizing( CurZoneEqNum ).ZoneTempAtHeatPeak);
+				this->cpSystemAir =  PsyCpAirFnWTdb (	DataSizing::FinalZoneSizing( CurZoneEqNum ).DesHeatCoilInHumRatTU ,
+														DataSizing::FinalZoneSizing( CurZoneEqNum ).DesHeatCoilInTempTU );
+				this->qDotZoneReq = FinalZoneSizing( CurZoneEqNum ).DesHeatLoad;
+				this->qDotZoneToHeatSetPt = FinalZoneSizing( CurZoneEqNum ).DesHeatLoad;
+				this->airAvailable = true;
+				this->heatingAvailable = true;
+				this->coolingAvailable = false;
+				SolveRegulaFalsi( ErrTolerance, 50, SolFlag, mDotAirSolutionHeating, std::bind( &HVACFourPipeBeam::residualSizing, this, _1  ), 0.0, maxFlowHeat );
+				if ( SolFlag == -1 ) {
+					ShowWarningError( "Heating load sizing search failed in four pipe beam unit called " + this->name );
+					ShowContinueError( "  Iteration limit exceeded in calculating size for design heating load" );
+				} else if ( SolFlag == -2 ) {
+					ShowWarningError( "Heating load sizing search failed in four pipe beam unit called " + this->name );
+					ShowContinueError( "  Bad size limits" );
+				}
+			}
+
+			// take the larger of heating and cooling
+			this->mDotDesignPrimAir = std::max(mDotAirSolutionHeating, mDotAirSolutionCooling );
+			// make sure this is higher than the zone OA requirement
+			this->mDotDesignPrimAir = std::max(this->mDotDesignPrimAir, FinalZoneSizing( CurZoneEqNum ).DesOAFlow * DataEnvironment::StdRhoAir);
+			this->vDotDesignPrimAir = this->mDotDesignPrimAir/ DataEnvironment::StdRhoAir;
+			this->totBeamLength = this->vDotDesignPrimAir / this->vDotNormRatedPrimAir;
+			if ( this->vDotDesignCWWasAutosized ) {
+				this->vDotDesignCW = this->vDotNormRatedCW * this->totBeamLength;
+			}
+			if ( vDotDesignHWWasAutosized ) {
+				this->vDotDesignHW = this->vDotNormRatedHW * this->totBeamLength;
+			}
+		}
+		// fill in mass flow rate versions of working variables (regardless of autosizing)
+		this->mDotDesignPrimAir = this->vDotDesignPrimAir * DataEnvironment::StdRhoAir;
+
+		if ( ( originalTermUnitSizeMaxVDot > 0.0 ) && ( originalTermUnitSizeMaxVDot != this->vDotDesignPrimAir ) && ( CurZoneEqNum > 0 ) ) {
+			// perturb system size to handle chnage in system size calculated without knowing about 4 pipe beam 
+			DataSizing::FinalSysSizing( DataZoneEquipment::ZoneEquipConfig(CurZoneEqNum).AirLoopNum ).DesMainVolFlow 
```

Comment:
@EnergyArchmage Here you can just do...

```CPP
DataSizing::FinalSysSizing( DataZoneEquipment::ZoneEquipConfig(CurZoneEqNum).AirLoopNum ).DesMainVolFlow += 
this->vDotDesignPrimAir - originalTermUnitSizeMaxVDot;
```
instead which is less code, but also better since it avoids another lookup.

You can do this for the following few sizing calls.